abstract-bits-derive/src/model.rs

```diff
@@ -260,43 +311,42 @@ fn controls_option(field: &syn::Field) -> Option<Ident> {
         .find(|a| a.path().is_ident("abstract_bits"))?;
 
     match parse(attr)? {
-        Ok(ident) => Some(ident),
+        Ok(expr) => Some(expr),
         Err(_) => abort!(attr.span(), "invalid abstract_bits attribute"; 
-            help = "The syntax is: #[abstract_bits(presence_of = <ident>)] with ident \
-            a later option type field"),
+            help = "The syntax is: #[abstract_bits(length_from = <expr>)] with expr \
+            an expression controlling this vec's length"),
     }
 }
 
-fn controls_list(field: &syn::Field) -> Option<Ident> {
-    fn parse(attr: &Attribute) -> Result<Ident, ()> {
+fn presence_from_attr(field: &syn::Field) -> Option<syn::Expr> {
+    fn parse(attr: &Attribute) -> Option<Result<syn::Expr, ()>> {
         let Ok(list) = attr.meta.require_list() else {
-            return Err(());
+            return Some(Err(()));
         };
         let mut tokens = list.tokens.clone().into_iter();
         match tokens.next() {
-            Some(TokenTree::Ident(ident)) if ident == "length_of" => (),
-            _ => return Err(()),
+            Some(TokenTree::Ident(ident)) if ident == "presence_from" => (),
+            _ => return None,
         }
         match tokens.next() {
             Some(TokenTree::Punct(punct)) if punct.as_char() == '=' => (),
-            _ => return Err(()),
+            _ => return Some(Err(())),
         }
-        let Some(TokenTree::Ident(target_field)) = tokens.next() else {
-            return Err(());
-        };
-        Ok(target_field)
+        
+        // Parse remaining tokens as expression
+        Some(syn::parse2::<syn::Expr>(tokens.collect()).map_err(|_| ()))
     }
 
     let attr = field
         .attrs
         .iter()
         .find(|a| a.path().is_ident("abstract_bits"))?;
 
-    match parse(attr) {
-        Ok(ident) => Some(ident),
+    match parse(attr)? {
+        Ok(expr) => Some(expr),
         Err(_) => abort!(attr.span(), "invalid abstract_bits attribute"; 
-            help = "The syntax is: #[abstract_bits(length_of = <ident>)] with ident \
-            a later option type field"),
+            help = "The syntax is: #[abstract_bits(presence_from = <expr>)] with expr \
+            a field expression controlling this option's presence"),
```

Comment:
I'm not sure of this help text but I can not think of anything better at the moment either. Maybe I'll think of something later